frontend/Components/Header.tsx

```diff
@@ -2,23 +2,18 @@ import * as React from 'react';
 import {
 	Link,
 } from 'react-router-dom';
-export interface HeaderParams{
-	pageName: string;
-}
-export class MainHeader extends React.Component<HeaderParams> {
-	private readonly pageName: string;
 
-	constructor(params: HeaderParams) {
-		super(params);
-		this.pageName = params.pageName;
-	}
+export class Header extends React.Component {
+	public state ={
```

Comment:
I am susprised ESLint didn't catch the lack of space between `=` and `{`.

If you want to declare `state` as a property I'd prefer it to be private, because it should never be accessed from the outside. I personally prefer to set the value of properties in the constructor rather than in the fields, because of two reasons - I want to have only one place to look at when determining default values and consistency.
You may have a situation where some of the fields always start with some constant default value, while others are set from properties. If you did it the way you did it here, you'd have to look both at the field declarations and constructor to know what value is where. But if everything is the constructor there's only one place to look.
Which segways into the consistency. if you always set the initial values in the constructor you never have to stop and think where to look for those default, because you **know** they're in the constructor. It's one of those small things that allows you to hold more things in your head at once.

Granted it's very arguable. I know some experienced developer who prefer various styles, but that's my personal favorite that I can actually give strong argumentation for :).

In other words, what I suggest:
 1. Remove the declaration of `state` field, it's not needed because it is already declared in `React.Component`
 2. Set the default value in the constructor.